source/validate.cpp

```diff
@@ -185,6 +185,11 @@ spv_result_t spvValidate(const spv_const_context context,
                                 binary->wordCount, setHeader,
                                 ProcessInstruction, pDiagnostic));
 
+  // Finish analyzing functions, in case a final OpFunctionEnd was not supplied.
+  for (auto& f : vstate.get_functions()) {
+    f.RegisterFunctionEnd();
+  }
```

Comment:
It would be better to call this in ValidationState_t::RegisterFunctionEnd(). I don't see anything in the spec that allows you to end a function without OpFunctionEnd.